probe-rs/src/error.rs

```diff
@@ -45,6 +45,9 @@ pub enum Error {
     /// the other architectures later.
     #[error("This capability has not yet been implemented for this architecture: {0}")]
     NotImplemented(&'static str),
+    /// The memory range request was not found in the memory interface provide.
+    #[error("No available memory at address {0:#x} of size {1:#x}")]
+    MemoryRangeNotFound(u64, u64),
```

Comment:
I found this helpful while developing the tests, because "most variables" passed the test, and there were only a few that required additional data in the coredump. Now that I've figured out how the coredumps work, I totally agree ... this should just panic and fail the test. I will make it so.

Reply:
Removed

Reply:
@Tiwalun ... after doing some more testing, I would like to suggest a compromise ...
- I've removed the error variant as you requested.
- Instead of a panic, I use an `Error::Other`. The reason for this is that in my tests, this should happen for "few of many" variables, and when it does, it is useful to see the error in the context of the variable tree. If it just panics, it makes it really difficult to understand which variable is affected.